### PR TITLE
genapi: Add support for TagsTagKey

### PR DESCRIPTION
**Pull request by justinsb**
Also fix up Update method to match Create method.

### CODE REVIEW

**Comment by justinsb** [comment]
I'm going to rebase this, but 

/assign @acpana

File: pkg/controller/direct/directbase/interfaces.go [hunk 1]
```diff
@@ -46,5 +46,5 @@ type Adapter interface {
 
 	// Update updates an existing GCP object.
 	// This should only be called when Find has previously returned true.
-	Update(ctx context.Context) (*unstructured.Unstructured, error)
+	Update(ctx context.Context, u *unstructured.Unstructured) error
```
Comment (acpana) [inline pkg/controller/direct/directbase/interfaces.go:49]: just to highlight, we are expecting that implementors will use `u` to write the status to it, right?

File: pkg/controller/direct/resourcemanager/tagkey_controller.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	}
+
+	a.actual = tagKey
+
+	return true, nil
+}
+
+// Delete implements the Adapter interface.
+func (a *tagKeyAdapter) Delete(ctx context.Context) (bool, error) {
+	// Already deletd
```
Comment (acpana) [inline pkg/controller/direct/resourcemanager/tagkey_controller.go:122]:
```suggestion
	// Already deleted
```

File: pkg/controller/direct/resourcemanager/tagkey_controller.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	req := &pb.DeleteTagKeyRequest{
+		Name: a.fullyQualifiedName(),
+	}
+
+	op, err := a.tagKeysClient.DeleteTagKey(ctx, req)
+	if err != nil {
+		if IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("deleting tagKey: %w", err)
```
Comment (acpana) [inline pkg/controller/direct/resourcemanager/tagkey_controller.go:137]: nit: here and elsewhere where it makes sense, do we want to also include the `tagKey` name? Assuming it doesn't bubble up in the direct controller.
```suggestion
		return false, fmt.Errorf("deleting tagKey %s: %w", a.fullyQualifiedName(), err)
```

**Review by acpana — COMMENTED** [review]
/hold for author to merge
/lgtm
/approve

Thanks for rebasing this 🙏🏼 💯 

I'd ignore the `license-lint` failure:

1) Fixed it in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1534
2) Verified it's not a real issue locally (ran w the fix).

File: pkg/controller/direct/directbase/interfaces.go (hunk 1 above)
Comment (justinsb) [reply pkg/controller/direct/directbase/interfaces.go:49]: Right.  Specifically we write to the Status.  I think in another PR I added more docs, I'll try to track those down!

**Comment by acpana** [comment]
/lgtm
/approve

thanks for addressing comments

**Pull request merged**